install/InstallHalyard.sh.gen

```diff
@@ -388,14 +389,23 @@ install_halyard
 
 configure_bash_completion
 
-start halyard
-
 printf 'Waiting for the Halyard daemon to start running'
 
+WAIT_START=$(date +%s)
+
 set +e 
 hal --ready &> /dev/null
 
 while [ "$?" != "0" ]; do
+  WAIT_NOW=$(date +%s)
+  WAIT_TIME=$(( $WAIT_NOW - $WAIT_START ))
+
+  if [ "$WAIT_TIME" -gt "$HALYARD_STARTUP_TIMEOUT_SECONDS" ]; then
```

Comment:
String comparison is lexicographic, and bash variables are untyped. This does an integer comparison.

Why do you prefer counting iterations?